### PR TITLE
Adds THEME_COLORS, POSTS_CATEGORIES, POSTS_CATEGORY_*, and HUSL color functions

### PR DESCRIPTION
**Pull request by Aeyoun**
See Issue #1980 before proceeding.

Colors are not displayed anywhere just yet, except by adding a ``<meta name="theme-color">``.

### CODE REVIEW

File: nikola/conf.py.in [hunk 1]
```diff
@@ -89,6 +89,10 @@ NAVIGATION_LINKS = ${NAVIGATION_LINKS}
 # Name of the theme to use.
 THEME = ${THEME}
 
+# Primary color of your theme. This will be used to customize your theme and
+# auto-generate related colors in POSTS_CATEGORY_COLORS. Must be a HEX value.
+THEME_COLOR = '#5670d4'
```
Comment (Kwpolska) [inline nikola/conf.py.in]: why this color?

Comment (Aeyoun) [inline nikola/conf.py.in]: Generic light corporate blue were designed to be non-offensive and enduce trust. Sounds like a perfect candidate for a nonintrusive default if I ever heard one. Goes well with the default bootstrap3 theme. Feel free to suggest another.

**Comment by Aeyoun** [comment]
@ralsina, @Kwpolska what do you guys think?

File: nikola/conf.py.in [hunk 2]
```diff
@@ -208,6 +212,57 @@ COMPILERS = ${COMPILERS}
...
+# Setting this to False generates a list page instead of an index. Indexes
+# are the default and will apply GENERATE_ATOM if set.
+# POSTS_CATEGORY_ARE_INDEXES = True
+
+# Each post and category page will have an associated color that can be used
+# to style them with a recognizable color detail across your site. A color
+# is assigned to  each category based on shifting the hue of your THEME_COLOR
+# at least 7.5 % while leaving the lightness and saturation untouched in the
+# HUSL colorspace. You can overwrite colors by assigning them colors in HEX.
+POSTS_CATEGORY_COLORS = {
```
Comment (Kwpolska) [inline nikola/conf.py.in]: should be commented out by default, along with POST_CATEGORY_DESCRIPTIONS and POST_CATEGORY_TITLE

File: nikola/data/themes/base-jinja/templates/categoryindex.tmpl [hunk 3]
```diff
@@ -0,0 +1,21 @@
+{#  -*- coding: utf-8 -*- #}
+{% extends 'index.tmpl' %}
+
+{% block extra_head %}
+    {{ super() }}
+    {% if generate_atom %}
+        <link rel="alternate" type="application/atom+xml" title="Atom for the {{ posts[0].category_name() }} section" href="/FIXME">
```
Comment (Kwpolska) [inline nikola/data/themes/base-jinja/templates/categoryindex.tmpl]: RSS?

File: nikola/data/themes/base/messages/messages_en.py [hunk 4]
```diff
@@ -36,4 +36,6 @@
     "Write your post here.": "Write your post here.",
     "old posts, page %d": "old posts, page %d",
     "page %d": "page %d",
+    "uncategorized": "Uncategorized",
```
Comment (Kwpolska) [inline nikola/data/themes/base/messages/messages_en.py]: both sides should be the same (uppercase)

File: nikola/utils.py [hunk 5]
```diff
@@ -1673,6 +1674,31 @@ def escape(s):
     return '/'.join([escape(p) for p in category_path])
 
 
+def colorize_str_from_base_color(string, base_color):
+    """Find a perceptual similar color from a base color based on the hash of a sring.
```
Comment (Kwpolska) [inline nikola/utils.py]: string*

also, it would be nice to expose this function as part of the global context

**Comment by Aeyoun** [comment]
I think it should be ready to merge now.

**Comment by Aeyoun** [comment]
@ralsina have you got any time to look through this before merging?

**Comment by ralsina** [comment]
Tomorrow morning I will.

**Comment by ralsina** [comment]
LGTM. However, since sections is a new concept it needs documenting. I can do it myself since you already did the hard work of actually implementing them :-)

**Pull request merged**